api/clients/eigenda_client.go

```diff
@@ -229,18 +252,41 @@ func (m *EigenDAClient) putBlob(ctx context.Context, rawData []byte, resultChan
 
 	alreadyWaitingForDispersal := false
 	alreadyWaitingForConfirmationOrFinality := false
+	var latestBlobStatus grpcdisperser.BlobStatus
 	for {
 		select {
 		case <-ctx.Done():
-			errChan <- fmt.Errorf("timed out waiting for EigenDA blob to confirm blob with request id=%s: %w", base64RequestID, ctx.Err())
+			// We can only land here if blob is still in
+			// 1. processing or dispersing status: waiting to land onchain
+			// 2. or confirmed status: landed onchain, waiting for finalization
+			// because all other statuses return immediately below.
+			//
+			// Assuming that the timeout is correctly set (long enough to both land onchain + finalize),
+			// 1. means that there is a problem with EigenDA, so we return an ErrorFailover to let the batcher failover to ethda
+			// 2. means that there is a problem with Ethereum, so we return 500.
```

Comment:
500, or 504. Probably more accurate to point out the grpc error

Reply:
oh ya good catch: [2912230](https://github.com/Layr-Labs/eigenda/pull/828/commits/29122300ece440a9269756fdb9c41b07545f77f4)